stdlib/ephemeron.ml

```diff
@@ -450,6 +512,18 @@ module K1 = struct
   let check_data (t:('k,'d) t) : bool = ObjEph.check_data t
   let blit_data (t1:(_,'d) t) (t2:(_,'d) t) : unit = ObjEph.blit_data t1 t2
 
+  let make key data =
+    let eph = create () in
+    set_key eph key;
+    set_data eph data;
+    eph
```

Comment:
It is safer in theory to write this as `set_data eph data; set_key eph key`, because if this is the last use of `key` and the GC runs between `set_key` and `set_data` then it can clear the ephemeron before the data is set, resulting in an ephemeron with a dead, cleared `key` and a strong reference to `data`. This strong reference will never be cleared.

I think in practice this situation will not occur, because there is no allocation between those two calls, the safepoints patch does not insert allocations at that point, and if this is inlined into a function that does contain allocations it seems relatively unlikely that the optimiser will choose to move an allocation between these calls. So it's probably fine as-is, but it would at least be more obviously fine the other way around.

([See here](https://github.com/ocaml-multicore/ocaml-multicore/issues/88#issuecomment-401799601) for further discussion on this issue. I think the new API proposed here is a big improvement, because it avoids this sort of trickiness)

Reply:
Doesn't cost much to fix, so I fixed it.